bundles/org.openhab.voice.googletts/src/main/java/org/openhab/voice/googletts/internal/GoogleTTSService.java

```diff
@@ -332,6 +340,38 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
         if (audio == null) {
             throw new TTSException("Could not read from Google Cloud TTS Service");
         }
-        return new ByteArrayAudioStream(audio, requestedFormat);
+
+        // compute the real format returned by google
+        AudioFormat realAudioFormat = parseAudioFormat(audio);
+
+        return new ByteArrayAudioStream(audio, realAudioFormat);
+    }
+
+    private AudioFormat parseAudioFormat(byte[] audio) {
+        try (InputStream inputStream = new ByteArrayInputStream(audio);
+                AudioInputStream audioInputStream = AudioSystem.getAudioInputStream(inputStream);) {
+            javax.sound.sampled.AudioFormat format = audioInputStream.getFormat();
+            String javaSoundencoding = format.getEncoding().toString();
+            String codecPCMSignedOrUnsigned;
+            if (javaSoundencoding.equals(Encoding.PCM_SIGNED.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_SIGNED;
+            } else if (javaSoundencoding.equals(Encoding.PCM_UNSIGNED.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_UNSIGNED;
+            } else if (javaSoundencoding.equals(Encoding.ULAW.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_ULAW;
+            } else if (javaSoundencoding.equals(Encoding.ALAW.toString())) {
+                codecPCMSignedOrUnsigned = AudioFormat.CODEC_PCM_ALAW;
+            } else {
+                codecPCMSignedOrUnsigned = null;
```

Comment:
In case codecPCMSignedOrUnsigned  is null (not identified), do you think your computations of bitRate and frequency are relevant ?
Or should you consider the default format in this case ?

Reply:
I honestly don't know.
If the codec is not recognized, or not one of the few mentionned here, the probability that the sink will play the sound without errors is pretty low. The value computed, or the default one, won't matter in the end.
I suppose it depends on the sink responsible for playing the sound and it's robustness.
Do you have a better educated guess ?